mail/postfix/Makefile

```diff
@@ -162,13 +167,12 @@ endif
 
 CCARGS+=-DDEF_DB_TYPE=\"$(default_database_type)\"
 
-config_directory=/etc/postfix# also add this to postfix init file
+# prevent postfix-install from executing postconf at build time
+# by explicitly providing the default values to postfix-install
+config_directory=/etc/postfix
 sample_directory=/etc/postfix
 command_directory=/usr/sbin
 daemon_directory=/usr/libexec/postfix
-data_directory=/usr/var/lib/postfix
-queue_directory=/usr/var/spool/postfix
-mail_spool_directory=/usr/var/mail
```

Comment:
/var is indeed symlinked to /tmp, but /usr/var is not. Why do you think /data would be a better place for these directories?

Reply:
AFAIK there is currently no convention about the mount point for persistent data in LEDE/OpenWrt. Daniel Golle [suggested](http://lists.infradead.org/pipermail/lede-dev/2016-September/002904.html) that /data should be used for persistent storage, like on Android.

According to the [Linux Documentation Project](http://www.tldp.org/LDP/Linux-Filesystem-Hierarchy/html/c23.html), /usr is typically used for "... the user binaries, their documentation, libraries, header files, ..."

/usr/var is not a good place for storing persistent data files because it is within /usr. For a LEDE/OpenWrt installation with no external storage, /usr/var is no better than /etc/postfix.

IMO /data is a good mount point for an external storage device because it makes it clear and highly visible that all variable data (what would normally be in /var) resides there. Even where there is no external storage device, placing all persistent data files into /data makes it easier to identify such files and include them into backup lists. /usr/var is an obscure and non-intuitive location for persistent data.

Reply:
While I admit /usr/var is not perfect, but /data looks too user-specific even though android has one. I suggest that we don't introduce a new top level directory yet, at least until LEDE/OpenWrt developers decide on the proper directory for persistent storage, or at least we'll see other LEDE/OpenWRT packages using /data. I think it is better to continue this discussion on mailing list. I agree on your points that /data might be better, but I would like to here the other opinions. Also, I can't find any replies on Daniel's mail, whether his suggestion is accepted. So, let's not change the default directories for now. Users are free to change them as they wish in their configuration.

Reply:
Okay, I will revert to using /usr/var for now. I am going to ask for feedback on the ML. IMHO we should have a convention on the mount point for a persistent storage device. We can come back to this issue after the discussion.

Reply:
As force-push of the amended commit erased some comments, I am copying and pasting from email notification:

@Shulyaka wrote:

> Please keep the directories as they were, i.e. $(persistent_data)/lib/postfix. I don't want to change the directories too often to avoid creating several unused directories on user devices.

I have restored the directory names in existence before this PR. As discussed earlier, we can come back to this issue once the lead developers express their opinion on the preferred mount point for the persistent data storage device.